pkg/controller/operators/catalog/operator.go

```diff
@@ -630,6 +630,15 @@ func (o *Operator) syncRegistryServer(logger *logrus.Entry, in *v1alpha1.Catalog
 
 	logger.Debug("ensured registry server")
 
+	// if catalog polling is enabled, and the polling interval is less than the default OLM sync cycle (15 minutes)
+	// requeue the catalog sync based on the polling interval, so the next check comes ahead of the default sync
+	if out.Spec.UpdateStrategy != nil && out.Spec.UpdateStrategy.Interval.Duration < queueinformer.DefaultResyncPeriod {
```

Comment:
I updated the logic to be more robust, PTAL

Basically it comes down to if the polling interval is <= the default, use that for requeueing. If it's greater, and enough time has passed since the last sync, use (polling interval % default) to get a sync ahead of the next cycle